hips/tiles/tests/test_tile.py

```diff
@@ -1,46 +1,87 @@
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.jpg'
+    def test_fetch_jpg(self, tmpdir):
+        meta = HipsTileMeta(order=3, ipix=463, file_format='jpg')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.jpg'
         tile = HipsTile.fetch(meta, url)
 
         assert tile.data.shape == (512, 512, 3)
-        assert_equal(tile.data[510][5:7], [[0, 0, 0], [1, 1, 1]])
+        assert_equal(tile.data[510][5:7], [[10, 10, 10], [11, 11, 11]])
 
-        filename = str(tmpdir / 'Npix30889.jpg')
+        filename = str(tmpdir / 'Npix463.jpg')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
         # The following assert fails, because on JPEG write / read
         # the data is different (for unknown reasons).
         # TODO: Figure out what's wrong here and fix!
-        # print(tile.data.sum())
+        # print(tile1.data.sum())
         # print(tile2.data.sum())
-        # print((tile == tile2).all())
+        # print((tile1 == tile2).all())
+        # assert tile1 == tile2
+
+    @remote_data
+    def test_fetch_png(self, tmpdir):
+        meta = HipsTileMeta(order=6, ipix=6112, file_format='png')
+        url = 'http://alasky.unistra.fr/2MASS6X/2MASS6X_H/Norder6/Dir0/Npix6112.png'
+        tile = HipsTile.fetch(meta, url)
+
+        assert tile.data.shape == (512, 512, 4)
+        assert_equal(tile.data[254][5:7], [[15, 15, 15, 255], [14, 14, 14, 255]])
+
+        filename = str(tmpdir / 'Npix6112.png')
+        tile.write(filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
+
+        assert tile == tile2
+
+    read_write_pars = [
+        dict(full_path='datasets/samples/DSS2Red/Norder3/Dir0/Npix463.fits', file_name='Npix463.fits',
+             file_format='fits', order=3, ipix=463, shape=(512, 512), tile_data=[3047, 3257]),
+        dict(full_path='datasets/samples/DSS2Red/Norder3/Dir0/Npix463.jpg', file_name='Npix463.jpg',
+             file_format='jpg', order=3, ipix=463, shape=(512, 512, 3), tile_data=[[10, 10, 10], [11, 11, 11]]),
+        dict(full_path='datasets/samples/2MASS6XH/Norder6/Dir0/Npix6112.png', file_name='Npix6112.png',
+             file_format='png', order=6, ipix=6112, shape=(512, 512, 4), tile_data=[[0, 0, 0, 0], [0, 0, 0, 0]]),
+    ]
+
+    @pytest.mark.parametrize('pars', read_write_pars)
+    @requires_hips_extra()
+    def test_read_write(self, tmpdir, pars):
+        full_path = get_hips_extra_file(pars['full_path'])
+        meta = HipsTileMeta(order=pars['order'], ipix=pars['ipix'], file_format=pars['file_format'])
+        tile = HipsTile.read(meta, full_path)
+
+        assert tile.data.shape == pars['shape']
+        assert_equal(tile.data[510][5:7], pars['tile_data'])
+
+        filename = str(tmpdir / pars['file_name'])
+        tile.write(filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
+
```

Comment:
Suggest to leave a TODO comment here, and fix the JPEG to round-trip in a later PR (like we discussed, by storing on `raw_data`)